### PR TITLE
Update keymap.c

### PR DESCRIPTION
**Pull request by crilith**
Updated to use included dynamic macro library and work with new hotswap socket board.

### CODE REVIEW

**Comment by drashna** [comment]
From #8008, you should upload a `rules.mk` file with basically just `DYNAMIC_MACROS_ENABLE = yes` in it, and to this branch.  

Specifically, any changes you make to the branch that you opened the PR from will automatically update this PR.  

Eg, https://github.com/crilith/qmk_firmware/tree/patch-3/keyboards/melody96/keymaps/crilith
Use the "Create new file" button, and do that there, and it will update this PR.

File: keyboards/melody96/keymaps/crilith/keymap.c [hunk 1]
```diff
@@ -30,19 +28,12 @@ const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
...
 
-const macro_t *action_get_macro(keyrecord_t *record, uint8_t id, uint8_t opt) {
-	return MACRO_NONE;
-}
-
 void matrix_init_user(void) {
 }
 
 void matrix_scan_user(void) {
 }
```
Comment (fauxpark) [inline keyboards/melody96/keymaps/crilith/keymap.c]: These functions can be deleted too :)

File: keyboards/melody96/keymaps/crilith/keymap.c [hunk 2]
```diff
@@ -30,19 +28,12 @@ const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
...
 
 bool process_record_user(uint16_t keycode, keyrecord_t *record) {
-    if (!process_record_dynamic_macro(keycode, record)) {
-        return false;
-    }
 	return true;
 }
```
Comment (fauxpark) [inline keyboards/melody96/keymaps/crilith/keymap.c]: And this.

File: keyboards/melody96/keymaps/crilith/keymap.c [hunk 3]
```diff
@@ -30,19 +28,12 @@ const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
...
-	return MACRO_NONE;
-}
-
 void matrix_init_user(void) {
 }
 
 void matrix_scan_user(void) {
 }
 
 bool process_record_user(uint16_t keycode, keyrecord_t *record) {
```
Comment (fauxpark) [inline keyboards/melody96/keymaps/crilith/keymap.c]:
```suggestion
```

File: keyboards/melody96/keymaps/crilith/keymap.c (hunk 2 above)
Comment (fauxpark) [inline keyboards/melody96/keymaps/crilith/keymap.c]:
```suggestion
```

**Pull request closed without merging**